src/adapters/messenger-adapter.js

```diff
@@ -111,51 +111,71 @@ class MessengerAdapter extends WebAdapter {
     res: express$Response,
   ): Promise<void> {
     logger.debug('handleRequest', req.body);
-    const messengerBody: MessengerBody = (req.body: any);
-    const { object, entry } = messengerBody;
-    if (object === 'page') {
-      for (const entryItem of entry) {
-        for (const event of entryItem.messaging) {
-          await this.processEvent(event); // eslint-disable-line no-await-in-loop
+    try {
+      const messengerBody: MessengerBody = (req.body: any);
+      const { object, entry } = messengerBody;
+      if (object === 'page') {
+        for (const entryItem of entry) {
+          for (const event of entryItem.messaging) {
+            await this.processEvent(event, entryItem.id); // eslint-disable-line no-await-in-loop
+          }
         }
+        res.sendStatus(200);
       }
-      res.sendStatus(200);
+    } catch (error) {
+      return res.status(400).send({ message: error.message, error });
```

Comment:
I have put the `try/catch` block here because there is no other place where we can catch error from messages validation and access the `res` object to send back the error status to messenger